### PR TITLE
#166214082 Database setup

### PR DESCRIPTION
**Pull request by Ukhu**
#### Description
Sets up Postgresql database and Sequelize.js ORM

#### Type of change
 - Install 'sequelize', 'sequelize-cli', 'pg' and 'pg-hstore' packages
 - Add .sequelizerc file
 - Bootstrap project using Sequelize-CLI to create necessary folders including; config, migrations, seeders, and models
 - Update default config file to use environment variables of the project
 - Add .env.sample file

#### How Has This Been Tested?
I tested the setup locally by using the Sequelize CLI to create a table in the database and populate it using dummy data from the seeding process.

#### Checklist:
N/A

#### Pivotal Tracker
#166214082

### CODE REVIEW

File: server/database/config/config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    password: process.env.DATABASE_PASSWORD,
+    database: process.env.DATABASE_TEST,
+    host: '127.0.0.1',
+    port: 5432,
+    dialect: 'postgres'
+  },
+  production: {
+    use_env_variable: 'DATABASE_URL'
+  }
+};
```
Comment (KvNGCzA) [inline server/database/config/config.js]: Add a new line here

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

File: server/database/config/config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  test: {
+    username: process.env.DATABASE_USERNAME,
+    password: process.env.DATABASE_PASSWORD,
+    database: process.env.DATABASE_TEST,
+    host: '127.0.0.1',
+    port: 5432,
+    dialect: 'postgres'
+  },
+  production: {
+    use_env_variable: 'DATABASE_URL'
```
Comment (KvNGCzA) [inline server/database/config/config.js:21]: Add `dialect: 'postgres'` here

File: server/database/config/config.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+require('dotenv').config();
+
+module.exports = {
+  development: {
+    username: process.env.DATABASE_USERNAME,
+    password: process.env.DATABASE_PASSWORD,
+    database: process.env.DATABASE_DEV,
+    host: '127.0.0.1',
+    port: 5432,
```
Comment (KvNGCzA) [inline server/database/config/config.js]: make port equals `process.env.DB_PORT`

File: server/database/config/config.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+require('dotenv').config();
+
+module.exports = {
+  development: {
+    username: process.env.DATABASE_USERNAME,
+    password: process.env.DATABASE_PASSWORD,
+    database: process.env.DATABASE_DEV,
+    host: '127.0.0.1',
```
Comment (KvNGCzA) [inline server/database/config/config.js]: make host equals `process.env.DB_HOST`

**Review by KvNGCzA — CHANGES_REQUESTED** [review]

**Review by Lundii — COMMENTED** [review]
I ran npx sequelize-cli init 
but I'm getting path must be a string error

**Comment by Ukhu** [comment]
> I ran npx sequelize-cli init
> but I'm getting path must be a string error

You're not supposed to run the 'npx sequelize-cli init' command again. That's basically like starting over the work I've done already.

File: package.json [hunk 5]
```diff
@@ -7,7 +7,8 @@
         "start": "node server/index.js",
         "start:dev": "nodemon --exec babel-node server/index.js",
         "test": "nyc mocha --timeout 10000 -r esm --exit",
-        "generate-lcov": "nyc report --reporter=text-lcov > lcov.info"
+        "generate-lcov": "nyc report --reporter=text-lcov > lcov.info",
+        "migrations": "npx sequelize db:migrate"
```
Comment (KvNGCzA) [inline package.json]: Remove this migration script and add the following
````
"db:rollback": "npm run db:back && npm run db:roll && npm run db:seed",
"db:roll": "sequelize db:migrate",
"db:back": "sequelize db:migrate:undo:all",
"db:seed": "sequelize db:seed:all"
````

**Comment by Lundii** [comment]
> > I ran npx sequelize-cli init
> > but I'm getting path must be a string error
> 
> You're not supposed to run the 'npx sequelize-cli init' command again. That's basically like starting over the work I've done already.

I ran the script using --force flag and it worked. I reverted the changes though.
I was able to create migrations, run it, create a database, create seeds and added seeds to my database using the link you gave me http://docs.sequelizejs.com/manual/migrations.html


I've not used sequelize before now but I guess these are the basic things to show that the configuration is working.

**Review by Lundii — APPROVED** [review]

**Review by KvNGCzA — APPROVED** [review]

**Review by allebd — APPROVED** [review]
Good

**Review by Nennyfills — APPROVED** [review]

**Review by Nennyfills — APPROVED** [review]
good job

**Pull request merged**